### PR TITLE
Added zoomable graphs

### PR DESCRIPTION
**Pull request by laf**
This is for the single large graphs only at the moment.

The JS code isn't 100% calculating the window correctly but I've got it to where I can, a fresh pair of eyes might help.

Having said that, it does work and is useable but has some quirks - please don't raise this to me direct.

Covers #1091

### CODE REVIEW

File: html/includes/functions.inc.php [hunk 1]
```diff
@@ -336,13 +336,16 @@ function print_graph_tag($args)
 
 function generate_graph_tag($args)
 {
+  if ($args['zoom'] == 'enabled') {
+      $g_id = 'id="317330dc6337227faa5df8dd149c344b"';
```
Comment (f0o) [inline html/includes/functions.inc.php]: what is this?

**Comment by laf** [comment]
The graph needs an id to be referenced, longer term we can support zooming on multiple graphs on a single page but as we only show one large graph per page at the moment I've just set a random id to the image id.

**Comment by f0o** [comment]
Right, although it doesnt matter for now, we should have this ID changed to something more reasonable in future, specially when we're dealing with multiple graphs on the same page.

I wonder though, isnt a jQuery selector sufficient for that? Then a CSS-Class like 'graph-zoomable' would be much nicer.

I'm giving this PR a spin in my test and report back in a bit

File: Makefile [hunk 2]
```diff
@@ -45,4 +45,4 @@ vis:
 	$(GIT_SUBTREE) --prefix=lib/vis https://github.com/almende/vis.git master
 
 typeahead:
```
Comment (f0o) [inline Makefile]: Since you touched this file, can you add typehead to the `update-subtrees` target as well? :)

**Comment by f0o** [comment]
Who wrote zoom.js ?

File: html/pages/graphs.inc.php [hunk 3]
```diff
@@ -200,7 +200,17 @@ function submitCustomRange(frmdata) {
...
   echo('<div style="width: '.$graph_array['width'].'; margin: auto;">');
+  echo '<div id="timepopup" style="position:absolute; display:none; z-index:1000;background:#FFFCBC;border:1px solid #c0c0c0"></div>';
+  echo "<div id='zoomBox' style='position:absolute; overflow:none; left:0px; top:0px; width:0px; height:0px; border:1px dashed #000; display:none; background:red; filter:alpha(opacity=50); -moz-opacity:0.5; -khtml-opacity:0.5; opacity:0.5'></div>";
+  $graph_array['zoom'] = 'enabled';
   echo(generate_graph_tag($graph_array));
+    echo "<script>
+                var img_src = $('#317330dc6337227faa5df8dd149c344b').attr('src');
+                var from = getUrlParameter('from',img_src);
+                var to = getUrlParameter('to',img_src);
+                initGraph($('#317330dc6337227faa5df8dd149c344b'),'graph.php?device='+img_src,from,to);
```
Comment (f0o) [inline html/pages/graphs.inc.php]: the passed base_url here contains from/to params so each zoom will actually duplicate these params.

**Pull request closed without merging**